content/docs/concepts/architecture.md

```diff
@@ -48,63 +64,47 @@ The contract of a {{<color "#6A00FF" >}}vWire{{< /color >}} is:
 
 ![Request](/img/concepts/architecture/request.svg)
 
 ### Close
-A {{<color "#6A00FF" >}}vWire{{< /color >}} between a {{<color "#0050EF" >}}Client{{< /color >}} and a Network Service is formally Closed by sending a 'Close' GRPC call to NSM.
+A {{<color "#6A00FF" >}}vWire{{< /color >}} between a {{<color "#0050EF" >}}Client{{< /color >}} and a Network Service is formally Closed by sending a 'Close' gRPC call to NSM.
 
 
 ### Monitor
 A {{<color "#6A00FF" >}}vWire{{< /color >}} between a {{<color "#0050EF" >}}Client{{< /color >}} and a Network Service always has a finite expire time.  The {{<color "#0050EF" >}}Client{{< /color >}} may (and usually does) send new 'Request' messages
 to 'refresh' the {{<color "#6A00FF" >}}vWire{{< /color >}}.  If a {{<color "#6A00FF" >}}vWire{{< /color >}} exceeds its expire time without being refreshed, NSM cleans up the {{<color "#6A00FF" >}}vWire{{< /color >}}.
 
-A {{<color "#0050EF" >}}Client{{< /color >}} may use a 'MonitorConnection' streaming GRPC call to NSM to get updates on the status of a {{<color "#6A00FF" >}}vWire{{< /color >}} it has to a Network Service.
+A {{<color "#0050EF" >}}Client{{< /color >}} may use a 'MonitorConnection' streaming gRPC call to NSM to get updates on the status of a {{<color "#6A00FF" >}}vWire{{< /color >}} it has to a Network Service.
 
-## Registries
+## Registry
 
-As with any other Mesh, Network Service Mesh has Network Service Registries (NSR) in which Network Services and Network Service {{<color "#008A00" >}}Endpoint{{< /color >}}s are registered
+The Network Service Registry (NSR) tracks Network Services and Network Service {{<color "#008A00" >}}Endpoints{{</color>}}. It implements a [simple gRPC API](https://github.com/networkservicemesh/api/blob/main/pkg/api/registry/registry.proto) that lets clients register, find, and unregister Network Services and {{<color "#008A00" >}}Endpoints{{</color>}}.
 
-### Network Service Endpoint
+There are two implementations currently: [in-memory volatile storage](https://github.com/networkservicemesh/cmd-registry-memory) and [Kubernetes custom resource storage](https://github.com/networkservicemesh/cmd-registry-k8s).
 
-A {{<color "#008A00" >}}Network Service Endpoint{{< /color >}} ({{<color "#008A00" >}}NSE{{< /color >}} or {{<color "#008A00" >}}Endpoint{{< /color >}}) provides one or more Network Services.  It registers with the registry a list
-of Network Services (by name) that it provides, and the 'destination labels' it is advertising for each Network Service.
+Clients and Endpoints don't connect directly to the Registry; they connect to the [Network Service Manager](https://github.com/networkservicemesh/cmd-nsmgr) which acts as a proxy for the registry. This simplifies NSM's internal data flows since Clients and Endpoints need to connect to the Manager anyway.
```

Comment:
```suggestion
In K8s Clients and Endpoints don't connect directly to the Registry; they connect to the [Network Service Manager](https://github.com/networkservicemesh/cmd-nsmgr) which acts as a proxy for the registry. This simplifies NSM's internal data flows since Clients and Endpoints need to connect to the Manager anyway.
```

Reply:
It might make more sense to move this into the K8s tabs, as its a feature of the K8s implementation of NSM, not a fundamental part of the base architecture.